internal/client/client.go

```diff
@@ -0,0 +1,188 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"foxygo.at/jig/pb/greet"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+type Client struct {
+	cc            *grpc.ClientConn
+	greeterClient greet.GreeterClient
+}
+
+func New(addr string) (*Client, error) {
+	cc, err := grpc.Dial(addr, grpc.WithInsecure())
+	if err != nil {
+		return nil, err
+	}
+	greet.NewGreeterClient(cc)
+	c := &Client{
+		cc:            cc,
+		greeterClient: greet.NewGreeterClient(cc),
+	}
+	return c, nil
+}
+
+func (c *Client) Close() error {
+	if c.cc != nil {
+		return c.cc.Close()
+	}
+	return nil
+}
+
+func (c *Client) Call(w io.Writer, names []string, streamType string) error {
+	if (streamType == "none" || streamType == "server") && len(names) != 1 {
```

Comment:
This checks for `"none"` but according to the switch below, `"unary"` is also acceptable. I'm not sure it makes sense to have both `"none"` and `"unary"` as synonyms. I would change the cli to change `none` to `unary` and just use a single name.